kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -107,8 +122,61 @@ instance SimplifyRuleLHS (OnePathRule Variable) where
 instance SimplifyRuleLHS (AllPathRule Variable) where
     simplifyRuleLhs = fmap (fmap AllPathRule) . simplifyRuleLhs . getAllPathRule
 
+instance SimplifyRuleLHS (RewriteRule Variable) where
+    simplifyRuleLhs = fmap (fmap RewriteRule) . simplifyRuleLhs' . getRewriteRule
+
 instance SimplifyRuleLHS (ReachabilityRule Variable) where
     simplifyRuleLhs (OnePath rule) =
         (fmap . fmap) OnePath $ simplifyRuleLhs rule
     simplifyRuleLhs (AllPath rule) =
         (fmap . fmap) AllPath $ simplifyRuleLhs rule
+
+
+simplifyRuleLhs'
+        :: forall simplifier variable
+        .  MonadSimplify simplifier
+        => SimplifierVariable variable
+        => RulePattern variable
+        -> simplifier (MultiAnd (RulePattern variable))
+simplifyRuleLhs' rule@(RulePattern _ _ _ _ _ _) = do
+        let simplifiedTerms = termToOrPattern left
+            rules =
+                map (setRuleLeft rule)
+                    (MultiOr.extractPatterns  simplifiedTerms)
+        return (MultiAnd.make rules)
+      where
+        RulePattern {left} = rule
+
+        setRuleLeft
+            :: RulePattern variable
+            -> Pattern variable
+            -> RulePattern variable
+        setRuleLeft
+            rulePattern
+            Conditional {term, predicate, substitution}
+          =
+            RulePattern.applySubstitution
+                substitution
+                rulePattern
+                    { RulePattern.left = term
+                    , RulePattern.requires = predicate
+                    }
+
+termToOrPattern
+    :: TermLike.InternalVariable variable
+    => TermLike variable
+    -> OrPattern variable
+termToOrPattern = Recursive.fold baseToOrPattern
+
+baseToOrPattern 
+    :: TermLike.InternalVariable variable
+    => Recursive.Base (TermLike variable) (OrPattern variable)
+    -> OrPattern variable
+baseToOrPattern (_ :< termLikeF) =
+    case termLikeF of
+        TermLike.OrF (TermLike.Or _ orFirst orSecond) ->
+            orFirst <> orSecond
+        TermLike.ApplySymbolF (TermLike.Application symbol children) ->
+            fromTermLike . TermLike.mkApplySymbol symbol . fmap toTermLike
+            <$> MultiOr.fullCrossProduct children
+        _ -> error $ show termLikeF
```

Comment:
in the tests presented in the [issue](https://github.com/kframework/kore/issues/699), this is always a `VariableF`; I'm not sure how to treat it

Reply:
Just add a new case for `TermLike.VariableF` that wraps it in an `OrPattern`.